src/lingrex/trimming.py

```diff
@@ -1,201 +1,178 @@
-        if seq[i] == "-":
-            gapped = True
-        else:
-            gapped = False
-        if gapped:
-            end += [i]
-        if not gapped:
-            break
-    return start[:-1], end[::-1][:-1]
-
-
-def gap_profile(alms, gap="-"):
-    """
-    Return a profile of the gap-ration per column.
-    """
-    return [col.count(gap) / len(col) for col in revert(alms)]
+    def soundclass(self, gap: str = GAP) -> str:
+        return token2class(self.first_sound(gap=gap) or "+", "cv")
 
 
-def trim_by_gap(alms, threshold=0.5, skeletons=("CV", "VC"), gap="-", exclude="_+"):
+class Sites(list):
     """
-    Trim alignment sites by gaps.
-
-    :parameter alms: Alignment sites of a cognate set.
-    :type alms: list
-    :parameter threshold: Threshold by which sites with gaps should be trimmed.
-        Defaults to '0.5'.
-    :type threshold: int
-    :param skeletons: Tuple of syllable-skeletons that should be preserved
-        for further processing. Defaults to '("CV", "VC")'.
-    :type skeletons: tuple
-    :parameter gap: String that codes gaps in alignment sites. Defaults to '-'.
-    :type gap: string
-    :param exclude: Sequence of strings that should be excluded from further processing,
-        e.g. morpheme boundaries. Defaults to '_+'.
-    :param exclude: str
-    :return: Indices of trimmed strings.
-    :rtype: set
+    A Sites object represents an alignment in the orthogonal view, i.e. listing columns rather
+    than rows.
     """
-    skeleton = get_skeleton(alms, gap=gap)
-    profile = gap_profile(alms, gap)
-    # exclude markers
-    idxs = [i for i, c in enumerate(skeleton) if c in exclude]
-    # order by gap weights
-    sorted_profile = sorted(enumerate(profile), key=lambda x: x[1], reverse=True)
-    while sorted_profile:
-        idx, score = sorted_profile.pop(0)
-        if score >= threshold:
-            current_skeleton = "".join([skeleton[i] for i in
-                                        range(len(skeleton)) if i not in idxs+[idx]])
+    def __init__(self, alms: typing.List[typing.Union[str, typing.List[str]]], gap: str = GAP):
+        """
+        :parameter alms: List of aligned sequences.
+        :parameter gap: String that codes gaps in alignment sites.
+        """
+        self.gap = gap
+        super().__init__(Site([row[i] for row in alms]) for i in range(len(alms[0])))
+
+    @property
+    def gap_ratios(self) -> typing.List[float]:
+        return [s.gap_ratio(gap=self.gap) for s in self]
+
+    @property
+    def soundclasses(self) -> typing.List[str]:
+        return [s.soundclass(gap=self.gap) for s in self]
+
+    def trimmed(self, idxs: typing.Iterable[int]) -> 'Sites':
+        for idx in sorted(set(idxs), reverse=True):
+            del self[idx]
+        return self
+
+    def to_alignment(self) -> typing.List[typing.List[str]]:
+        return [[s[i] for s in self] for i in range(len(self[0]))]
+
+    def trimmed_by_gap(self,
```

Comment:
The `trim_*` functions have been replaced by methods on `Sites` returning not lists of indices, but a trimmed `Sites` object.